arch/arm/src/nrf52/nrf52_i2c.c

```diff
@@ -223,6 +223,8 @@ static int nrf52_i2c_transfer(FAR struct i2c_master_s *dev,
   uint32_t regval = 0;
   int      ret = OK;
 
+  nxsem_wait_uninterruptible(&priv->sem_excl);
+
```

Comment:
Just a note:  Use of nxsem_wait_uninterruptible also make any task using this function to be non-cancelable.  See Issue #619

Reply:
I think it would be more correct to use:

    ret = nxsem_wait&priv->sem_excl);
    if (ret < 0)
      {
        return ret;
      }

Would you also want return if interrupted by a signal?  That is required by most POSIX read/write functions.

Reply:
In any case, should you not handle any error returned by nxsem_wait_uninterruptle.  It can return errors under certain conditions.

Reply:
Yes, you are right. I just made a copy-paste from stm32 which I see is also affected by this issue. Corrected as you recomended.

Reply:
> Yes, you are right. I just made a copy-paste from stm32 which I see is also affected by this issue. Corrected as you recomended.

The current nxsem_wait_unterruptible and nxsem_timedwait_uninterruptible have basically broken all task/thread cancellation.  The simple fix is to just don't loop on the ECANCELED error, but then all of the callers would have to handle the return value correctly.  Most, like the code that you leveraged from to do not.  So that logic is all doubly broken.  Unfortunately, there are many places where nxsem_[timed]wait_interruptible is called and so the fix is not simple.

In this case, you logic will now return if the task is canceled (which is good) or if any signal is received.  This latter might be good or bad, depending upon why nrf52_i2c_transfer() is called and what the caller will do with the error.  There could be cases where EINTR should be ignored(?)